201 CT11 Datastore

charles-turner-1 opened the pull request:
Made some progress grinding through all these observational datasets.

@rbeucher this is nowhere near ready but if you get the chance can you take a quick look & tell me whether this is roughly what you were after? 

Are there any docs/guides on what to do if we can't match things to the controlled vocabularies correctly? I've just been filling in with what seems sensible to me - is there a better way?

rbeucher commented:
I'm gonna have a look. The vocabulary should be CMIP vocabulary. Do you have a specific example?

charles-turner-1 commented:
This is probably the cleanest example:

```python
def split_filename_CERES_EBAF(file_path : str) -> dict[str, str | None]:
    """
    Take a filename & split it's stem on occurences of `_`. This
    returns a dictionary which we can then use to create a new 
    dataframe with all the various fields.

    These are of the form:
    `rlds_CERES-EBAF_L3B_Ed2-7_200003-201209` which get mapped as
    f{variable_id}_{source_id}_{experiment_id}_{version}_{time_range}
    or 
    `OBS_CERES-EBAF_sat_Ed4.0_Amon_rsut_200003-201812` which get mapped as
    f"{project_id}_{source_id}_{experiment_id}_{version}_{table_id}_{variable_id}_{time_range}"

    
    
    """
    # Remove the file extension and split the rest based on '_'
    components = Path(file_path).stem.split('_')
    # Return components in a dictionary format for DataFrame usage
    match components:
        case [variable_id, source_id, experiment_id, version, time_range]:
            return {
                'project_id': "observations",
                'source_id': source_id,
                'experiment_id': experiment_id,
                'version': version,
                'table_id': None,
                'variable_id': variable_id,
                'time_range': time_range
            }
        case [project_id, source_id, experiment_id, version, table_id, variable_id, time_range]:
            return {
                'project_id': project_id,
                'source_id': source_id,
                'experiment_id': experiment_id,
                'version': version,
                'table_id': table_id,
                'variable_id': variable_id,
                'time_range': time_range
            }
```

Some of these are easy enough to figure out - eg. `variable_id`, `time_range` etc etc are pretty obvious. But I've just filled in `project_id` as `'observations'` for the first name as there's nothing really obvious that matches it & I couldn't find any helpful information about what these should be mapped to - is there somewhere I want to look for this info?

rbeucher commented:
For the project_id, I agree it is a bit messy.

Files usually have the OBS or OBS6 prefix. That can be used as project_id.
Files without prefix are essentially Obs4MIPs

rbeucher commented:
It would be good too include the Tier1, Tier2, Tier3 in a separate column if possible.

rbeucher reviewed (changes requested):
Looks good to me. Add the project as OBS, OBS6, OBS4MIPS like I did in my example.
See if you can add the Tier information in a separate column. That might require having a translator in the access-nri-intake package though... I wanted to avoid that but not a big deal...

charles-turner-1 commented:
Okay, sounds good - I should be able to make that work.

charles-turner-1 committed:
coming along

flicj191 reviewed (approved):
The df's look good to me. Thanks

rbeucher commented:
Thanks @flicj191 .

@charles-turner-1 If you could add that to the catalog that would be great.
Eventually we will ask NCI to copy it to the ct11 project.

charles-turner-1 committed:
Update catalog.json file

charles-turner-1 committed:
Updates (keep these for posterity?)

charles-turner-1 commented:
Do we want to merge this in here or just close the PR?
